default.nix

```diff
@@ -63,7 +63,14 @@ assert (lib.assertMsg ((builtins.isPath doomPrivateDir)
   "doomPrivateDir must be either a path, a derivation or a stringified store path");
 
 let
-  flake = import ./flake-compat-helper.nix { src = ./.; };
+  flake =
+    (import
+      (let lock = with builtins; fromJSON (readFile ./flake.lock); in
+       builtins.fetchTarball {
+         url = "https://github.com/edolstra/flake-compat/archive/${lock.nodes.flake-compat.locked.rev}.tar.gz";
+         sha256 = lock.nodes.flake-compat.locked.narHash;
+       })
+      { src = ./.; }).defaultNix;
```

Comment:
This works because `defaultNix` is `result` plus a merge of `packages` or `defaultPackages` if they exist: https://github.com/edolstra/flake-compat/blob/master/default.nix#L191-L194